docs/modules/ROOT/pages/api/udc.adoc

```diff
@@ -0,0 +1,86 @@
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+
+= Universal Deployer
+
+Reference of the Universal Deployer Contract (UDC) interface and preset contract.
+
+== Core
+
+[.contract]
+[[IUniversalDeployer]]
+=== `++IUniversalDeployer++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.10.0/src/utils/universal_deployer/interface.cairo#L7[{github-icon},role=heading-link]
+
+```javascript
+use openzeppelin::utils::universal_deployer::interface::IUniversalDeployer;
+```
+
+[.contract-index]
+.Functions
+--
+* xref:#IUniversalDeployer-deploy_contract[`++deploy_contract(class_hash, salt, from_zero, calldata)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#IUniversalDeployer-ContractDeployed[`++ContractDeployed(address, deployer, from_zero, class_hash, calldata, salt)++`]
+--
+
+[#IUniversalDeployer-Functions]
+==== Functions
+
+[.contract-item]
+[[IUniversalDeployer-deploy_contract]]
+==== `[.contract-item-name]#++deploy_contract++#++(class_hash: ClassHash, salt: felt252, from_zero: bool, calldata: Span<felt252>) → ContractAddress++` [.item-kind]#external#
+
+Deploys a contract through the Universal Deployer Contract.
+
+[#IUniversalDeployer-Events]
+==== Events
+
+[.contract-item]
+[[IUniversalDeployer-ContractDeployed]]
+==== `[.contract-item-name]#++ContractDeployed++#++(address: ContractAddress, deployer: ContractAddress, from_zero: bool, class_hash: ClassHash, calldata: Span<felt252>, salt: felt252)++` [.item-kind]#event#
+
+Emitted when `deployer` deploys a contract through the Universal Deployer Contract.
+
+== Presets
+
+:origin-dependent: xref:/udc.adoc#origin_dependent[origin-dependent]
+:origin-independent: xref:/udc.adoc#origin_independent[origin-independent]
+
+[.contract]
+[[UniversalDeployer]]
+=== `++UniversalDeployer++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.10.0/src/presets/universal_deployer.cairo[{github-icon},role=heading-link]
+
+```javascript
+use openzeppelin::presets::UniversalDeployer;
+```
+
+The standard Universal Deployer Contract.
+
+include::../utils/_class_hashes.adoc[]
+
+[.contract-index]
+.{presets-page}
+--
+{UniversalDeployer-class-hash}
+--
+
+[.contract-index]
+.Embedded Implementations
+--
+.UniversalDeployerImpl
+* xref:#UniversalDeployer-deploy_contract[`++deploy_contract(self, address, deployer, from_zero, class_hash, calldata, salt)++`]
+--
+
+[.contract-item]
+[[UniversalDeployer-deploy_contract]]
+==== `[.contract-item-name]#++deploy_contract++#++(ref self: ContractState, address: ContractAddress, deployer: ContractAddress, from_zero: bool, class_hash: ClassHash, calldata: Span<felt252>, salt: felt252) -> ContractAddress++` [.item-kind]#external#
+
+Deploys a contract through the Universal Deployer Contract.
+
+When `from_zero` is `false`, the deployment type is {origin-dependent}.
```

Comment:
I think we should mention that the inner deploy_syscall is receiving a modified salt being a hash of the caller address and the received one.

The reason I'm insisting in this, is that UDC deployments are not the only way, users could use deploy_syscall as well, and deploy_syscall also have a from_zero, that has origin-dependet and independent approaches, that are not the same as UDC, and if we don't mention the UDC specific difference, this could be very confusing to users.